physicalplan/scan/literal_selector.go

```diff
@@ -63,6 +63,8 @@ func (o *numberLiteralSelector) Next(ctx context.Context) ([]model.StepVector, e
 	totalSteps := int64(1)
 	if o.step != 0 {
 		totalSteps = (o.maxt-o.mint)/o.step + 1
+	} else {
+		o.step = 1
```

Comment:
Not sure I understand what this is doing for all the selector `Next` methods. Is it to prevent further divide by zero errors in the future? 🙂

Reply:
We have to set the step to a non-zero value to make sure the operator terminates for instant queries. Otherwise we won't make any changes to the currentStep here:  https://github.com/thanos-community/promql-engine/blob/main/physicalplan/scan/vector_selector.go#L109

Reply:
Thanks!